slang.h

```diff
@@ -3051,6 +2837,541 @@ namespace slang
+            */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL enableReproCapture() = 0;
+
+            /** Get the (linked) program for a compile request.
+
+            The linked program will include all of the global-scope modules for the
+            translation units in the program, plus any modules that they `import`
+            (transitively), specialized to any global specialization arguments that
+            were provided via the API.
+            */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL getProgram(
+            slang::IComponentType** outProgram) = 0;
+
+            /** Get the (partially linked) component type for an entry point.
+
+            The returned component type will include the entry point at the
+            given index, and will be specialized using any specialization arguments
+            that were provided for it via the API.
+
+            The returned component will *not* include the modules representing
+            the global scope and its dependencies/specialization, so a client
+            program will typically want to compose this component type with
+            the one returned by `spCompileRequest_getProgram` to get a complete
+            and usable component type from which kernel code can be requested.
+            */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL getEntryPoint(
+            SlangInt                entryPointIndex,
+            slang::IComponentType** outEntryPoint) = 0;
+
+            /** Get the (un-linked) module for a translation unit.
+
+            The returned module will not be linked against any dependencies,
+            nor against any entry points (even entry points declared inside
+            the module). Similarly, the module will not be specialized
+            to the arguments that might have been provided via the API.
+
+            This function provides an atomic unit of loaded code that
+            is suitable for looking up types and entry points in the
+            given module, and for linking together to produce a composite
+            program that matches the needs of an application.
+            */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL getModule(
+            SlangInt                translationUnitIndex,
+            slang::IModule**        outModule) = 0;
+
+            /** Get the `ISession` handle behind the `SlangCompileRequest`.
+            TODO(JS): Arguably this should just return the session pointer.
+            */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL getSession(
+            slang::ISession** outSession) = 0;
+
+            /** get reflection data from a compilation request */
+        virtual SLANG_NO_THROW SlangReflection* SLANG_MCALL getReflection() = 0;
+
+            /** Make output specially handled for command line output */
+        virtual SLANG_NO_THROW void SLANG_MCALL setCommandLineCompilerMode() = 0;
```

Comment:
As discussed elsewhere, this is probably something we'd want to put into a hidden interface that is in a non-public header, so that `slangc` can perform a `queryInterface` and then toggle a given compile request into command-line mode without having users be able to do that and shoot themselves in the foot.

This still raises the issue that command-line compilation is fundamentally linked to what we call "end-to-end" compilation in the current API, and there are certain features that are only allowed in "end-to-end mode."